src/Admin/AbstractAdmin.php

```diff
@@ -511,7 +512,7 @@ abstract class AbstractAdmin implements AdminInterface, DomainObjectInterface, A
     private $show;
 
     /**
-     * @var Form|null
+     * @var FormInterface
```

Comment:
FormInterface|null

Reply:
it will never be null, since `$formBuilder->getForm()` returns `FormInterface` only

Reply:
$form property can still be null, even if the getter cannot return null.

Moreover getForm is currently returning ?FormInterface

Because it can be null if you (wrongly) call this method during the form building process. But we could add an execption for this case.

Reply:
IMO if there is any change for this to be null in any step of the Admin Creation (not set in the construct) we should add |null.

Reply:
@VincentLanglet 

> Moreover getForm is currently returning ?FormInterface

I don't see this code, where is it? I do see return type is `FormInterface`

----------------

anyway I'll set return type to be nullable

Reply:
I was looking on master indeed
https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Admin/AbstractAdmin.php#L1156
https://github.com/sonata-project/SonataAdminBundle/blob/master/src/Admin/AdminInterface.php#L112